back-end/yelpCrawler.py

```diff
@@ -0,0 +1,197 @@
+    """Given your API_KEY, send a GET request to the API.
+    Args:
+        host (str): The domain host of the API.
+        path (str): The path of the API after the domain.
+        API_KEY (str): Your API Key.
+        url_params (dict): An optional set of query parameters in the request.
+    Returns:
+        dict: The JSON response from the request.
+    Raises:
+        HTTPError: An error occurs from the HTTP request.
+    """
+    url_params = url_params or {}
+    url = '{0}{1}'.format(host, quote(path.encode('utf8')))
+    headers = {
+        'Authorization': 'Bearer %s' % api_key,
+    }
+    response = requests.request('GET', url, headers=headers, params=url_params)
+
+    return response.json()
+
+def search(api_key, term, latitude, longitude):
+    """Query the Search API by a search term and location.
+    Args:
+        term (str): The search term passed to the API.
+        location (str): The search location passed to the API.
+    Returns:
+        dict: The JSON response from the request.
+    """
+    url_params = {
+        'term': term.replace(' ', '+'),
+        'latitude': latitude,
+        'longitude': longitude,
+        'radius': RADIUS,
+        'open_now': OPEN_NOW.replace(' ', '+'),
+        'limit': SEARCH_LIMIT
+    }
+    return request(API_HOST, SEARCH_PATH, api_key, url_params=url_params)
+
+
+def get_business(api_key, business_id):
+    """Query the Business API by a business ID.
+    Args:
+        business_id (str): The ID of the business to query.
+    Returns:
+        dict: The JSON response from the request.
+    """
+    business_path = BUSINESS_PATH + business_id
+
+    return request(API_HOST, business_path, api_key)
+
+def getYelpData(lat=37.778163, long=-122.411908):
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('-s', '--term', dest='term', default=DEFAULT_TERM,
+                        type=str, help='Search term (default: %(default)s)')
+    parser.add_argument('-la', '--longitude', dest='longitude', default=LONGITUDE,
+                        type=float, help='Search longitude (default: %(default)f)')
+    parser.add_argument('-lo', '--latitude', dest='latitude', default=LATITUDE,
+                        type=float, help='Search latitude (default: %(default)f)')
+
+    input_values = parser.parse_args()
+
+    try:
+        response = search(API_KEY, input_values.term, lat, long)
+        businesses = response.get('businesses')
+        businessList = []
+        for idx in businesses:
+            response = get_business(API_KEY, idx['id'])
+            businessList.append(response)
+    except HTTPError as error:
+        sys.exit(
+            'Encountered HTTP error {0} on {1}:\n {2}\nAbort program.'.format(
+                error.code,
+                error.url,
+                error.read(),
+            )
+        )
+
+    # Merge loadJsonFile.py in this file
+    datainfo = businessList
+
+    datalist = []
+
+    for item in datainfo:
```

Comment:
This for loop is quite big - can we add comments here and there within this for loop?